File: engine/engine-model/src/main/scala/net/server/channel/packet/special/BaseSpecialMovePacket.scala
package net.server.channel.packet.special

import java.awt.Point

import net.server.MaplePacket

class BaseSpecialMovePacket(private var _skillId: Int, private var _skillLevel: Int, private var _position: Point) extends MaplePacket {
  def skillId: Int = _skillId

  def skillLevel: Int = _skillLevel

  def position: Point = _position
}

File: engine/engine-model/src/main/java/net/server/channel/packet/special/BaseSpecialMovePacket.java
package net.server.channel.packet.special;

import java.awt.Point;

import net.server.MaplePacket;

public class BaseSpecialMovePacket implements MaplePacket {
   private final Integer skillId;

   private final Integer skillLevel;

   private final Point position;

   public BaseSpecialMovePacket(Integer skillId, Integer skillLevel, Point position) {
      this.skillId = skillId;
      this.skillLevel = skillLevel;
      this.position = position;
   }

   public Integer skillId() {
      return skillId;
   }

   public Integer skillLevel() {
      return skillLevel;
   }

   public Point position() {
      return position;
   }
}
